src/ini.pest

```diff
@@ -1,16 +1,22 @@
-ini = { SOI ~ (pair | comment | NEWLINE)* ~ section* ~ EOI }
-WHITESPACE = _{ " " | "\t" }
+ini = { SOI ~ (pair | comment)* ~ section* }
+
+// lines
+pair = { key ~ "=" ~ value ~ eol}
+comment = { ("#" | ";") ~ value? ~ eol}
 
-// section
-section = { "[" ~ name ~ "]" ~ body }
-name = @{ char+ }
-char = { ASCII_ALPHANUMERIC | "." | "_" | "/" | " " }
-body = { (NEWLINE ~ (pair | comment)?)* }
+// sections
+section = { "[" ~ name ~ "]" ~ eol ~ body? }
```

Comment:
All these rules with `eol` in them &ndash; what happens if there's a section at the very end of the file, but without a final newline? These rules wouldn't match.

Reply:
`eol` matches `NEWLINE` or  `EOI` - so that would produce a empty section.

Reply:
I like the way this rule reads, but it simply doesn't work with a section like `[foo]bar\n`. I have to say, however, that this is a really dumb argument and I don't see why we should at all support `[foo]bar\n`. I would love to remove this from the tests.

Reply:
I also don't think we should support trailing whitespace after a section, as it serves no purpose (e.g., `[foo]    \n`).

Reply:
Unfortunately, this rule does not include spaces within a section name (e.g., `[  foo  ]`). And, technically, there's no reason why we shouldn't support this one, because you can absolutely name files with spaces in them. Try `touch "  foo"` and it will create a file named `  foo`.

I don't know how to fix this one without changing the `WHITESPACE` rule, unfortunately.